Batch window issue

ucfnet002 opened the pull request:
Pull Request for Issue # .
#### Summary of Changes

Moved ".modal-body {" into mediaquery "@media (max-width: 767px)" in less-file of "isis"-template
#### Testing Instructions

in a menu, or content window select some items then click batch attempt to "move or copy" you will find the window is too small.
Apply this patch
repeat the test and on a standard screen it overflows correctly to allow you to select an item.
If possible test on multiple view ports particularly if you have a device has the max-width as above.
Not sure if it would be better removing it from the constraint totally.

ucfnet002 committed:
Pull Request for Issue #9551 and #9586

Changing the .less instead of .css

infograf768 commented:
Have not tested but this patch is incomplete as generate.css has not been ran.
If I run php build/generatecss.php today, I get:
```
diff --git a/administrator/templates/isis/css/template-rtl.css b/administrator/templates/isis/css/template-rtl.css
index 99a336c..bc30e25 100644
--- a/administrator/templates/isis/css/template-rtl.css
+++ b/administrator/templates/isis/css/template-rtl.css
@@ -8171,8 +8171,8 @@
 		padding: 5px 0;
 	}
-}
-.modal-body {
-	-webkit-overflow-scrolling: touch;
-	overflow-y: auto !important;
+	.modal-body {
+		-webkit-overflow-scrolling: touch;
+		overflow-y: auto !important;
+	}
 }
 .pull-right {
diff --git a/administrator/templates/isis/css/template.css b/administrator/templates/isis/css/template.css
index ba425fb..bad66c4 100644
--- a/administrator/templates/isis/css/template.css
+++ b/administrator/templates/isis/css/template.css
@@ -2791,5 +2791,5 @@
 }
 .alert h4 {
-  margin: 0 0 .5em;
+	margin: 0 0 .5em;
 }
 .alert .close {
@@ -8171,7 +8171,7 @@
 		padding: 5px 0;
 	}
-}
-.modal-body {
-	-webkit-overflow-scrolling: touch;
-	overflow-y: auto !important;
+	.modal-body {
+		-webkit-overflow-scrolling: touch;
+		overflow-y: auto !important;
+	}
 }
diff --git a/administrator/templates/isis/less/template.less b/administrator/templates/isis/less/template.less
index 0b1a3d4..c5166af 100644
--- a/administrator/templates/isis/less/template.less
+++ b/administrator/templates/isis/less/template.less
@@ -1324,9 +1324,9 @@
 		padding: 5px 0;
 	}
-}
 
-/* Fix iOS scrolling inside bootstrap modals (using iframe) */
-.modal-body {
-	-webkit-overflow-scrolling: touch;
-	overflow-y: auto !important;
+	/* Fix iOS scrolling inside bootstrap modals (using iframe) */
+	.modal-body {
+		-webkit-overflow-scrolling: touch;
+		overflow-y: auto !important;
+	}
 }
diff --git a/templates/protostar/css/template.css b/templates/protostar/css/template.css
index cbc9db4..ff08400 100644
--- a/templates/protostar/css/template.css
+++ b/templates/protostar/css/template.css
@@ -2806,5 +2806,5 @@
 }
 .alert h4 {
-  margin: 0 0 .5em;
+	margin: 0 0 .5em;
 }
 .alert .close {
```

ucfnet002 commented:
sorry, didn't knew that. was my first submit to the joomla project.

brianteeman commented:
I made the same comment in #9586

ucfnet002 commented:
brian, i saw your comment and so i decided to make this pull request. maybe it was a wrong way?

wojsmol committed:
run generatecss.php

wojsmol commented:
@ucfnet002 @infograf768 see https://github.com/ucfnet002/joomla-cms/pull/1

andrepereiradasilva commented:
@wojsmol i think you made the patch in wrong branch (`ucfnet002:staging`) when it should be the `ucfnet002:patch-1` branch

ucfnet002 committed:
Merge pull request #3 from wojsmol/ucfnet-generetcss

run generatecss.php

wojsmol commented:
@aaleksanyants thank you - fixed

andrepereiradasilva commented:
I have tested this item :red_circle: unsuccessfully on 92df3544b065065b42035b3a070fc16d60f3b782<br /><br />Works fine on desktop and tablet but not on mobile viewport.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9683">issues.joomla.org/joomla-cms/9683</a>.</sub>

Twincarb commented:
The comment held in the less file is as below, I was to busy to amend and test a correct solution.

Fix iOS scrolling inside bootstrap modals (using iframe) */

The testing of this PR would also need to take this into consideration. Was to busy to update the previous PR, and to investigate the reason for the original change. 

Possible changes to the code will need to be :

/* ----------- iPhone 4 and 4S ----------- */

/* Portrait and Landscape */
@media only screen 
  and (min-device-width: 320px) 
  and (max-device-width: 480px)
  and (-webkit-min-device-pixel-ratio: 2) {

}

/* Portrait */
@media only screen 
  and (min-device-width: 320px) 
  and (max-device-width: 480px)
  and (-webkit-min-device-pixel-ratio: 2)
  and (orientation: portrait) {
}

/* Landscape */
@media only screen 
  and (min-device-width: 320px) 
  and (max-device-width: 480px)
  and (-webkit-min-device-pixel-ratio: 2)
  and (orientation: landscape) {

}

/* ----------- iPhone 5 and 5S ----------- */

/* Portrait and Landscape */
@media only screen 
  and (min-device-width: 320px) 
  and (max-device-width: 568px)
  and (-webkit-min-device-pixel-ratio: 2) {

}

/* Portrait */
@media only screen 
  and (min-device-width: 320px) 
  and (max-device-width: 568px)
  and (-webkit-min-device-pixel-ratio: 2)
  and (orientation: portrait) {
}

/* Landscape */
@media only screen 
  and (min-device-width: 320px) 
  and (max-device-width: 568px)
  and (-webkit-min-device-pixel-ratio: 2)
  and (orientation: landscape) {

}

/* ----------- iPhone 6 ----------- */

/* Portrait and Landscape */
@media only screen 
  and (min-device-width: 375px) 
  and (max-device-width: 667px) 
  and (-webkit-min-device-pixel-ratio: 2) { 

}

/* Portrait */
@media only screen 
  and (min-device-width: 375px) 
  and (max-device-width: 667px) 
  and (-webkit-min-device-pixel-ratio: 2)
  and (orientation: portrait) { 

}

/* Landscape */
@media only screen 
  and (min-device-width: 375px) 
  and (max-device-width: 667px) 
  and (-webkit-min-device-pixel-ratio: 2)
  and (orientation: landscape) { 

}

/* ----------- iPhone 6+ ----------- */

/* Portrait and Landscape */
@media only screen 
  and (min-device-width: 414px) 
  and (max-device-width: 736px) 
  and (-webkit-min-device-pixel-ratio: 3) { 

}

/* Portrait */
@media only screen 
  and (min-device-width: 414px) 
  and (max-device-width: 736px) 
  and (-webkit-min-device-pixel-ratio: 3)
  and (orientation: portrait) { 

}

/* Landscape */
@media only screen 
  and (min-device-width: 414px) 
  and (max-device-width: 736px) 
  and (-webkit-min-device-pixel-ratio: 3)
  and (orientation: landscape) { 

}<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9683">issues.joomla.org/joomla-cms/9683</a>.</sub>

ucfnet002 committed:
Try to fix #9683 with modal batch window

Reverted the last change and added overflow-y to .field-media-wrapper .modal .modal-body {}

ucfnet002 commented:
I'm a little confused about github, so i don't know how to pull my change ... here is a new bugfix. also i reverted the last changes: https://github.com/ucfnet002/joomla-cms/commit/eeb23d4f9746053ee7ff5d43437f6b63a448fe25

ucfnet002 committed:
Updated template.less and

andrepereiradasilva commented:
> I'm a little confused about github

the first time is normal.


bottom line is, when you change any file in your `patch-1` branch (the branch you originally used to create this pull request) in your `joomla-cms` fork repository they will automatically be in this PR.


this is your `patch-1` branch https://github.com/ucfnet002/joomla-cms/tree/patch-1

ucfnet002 committed:
2nd try to fix #9683 with modal batch window

Removed "overflow-y: auto !important;"

ucfnet002 commented:
I deleted line 1329 and it seems to work ... here is my " administrator/templates/isis/css/template-rtl.css
": https://github.com/ucfnet002/joomla-cms/commit/e4446c5f3f74943c10852f699e2739e5c3d2f240

wojsmol commented:
@ucfnet002 run `php build/generatecss.php` from command line

ucfnet002 commented:
My dev environment isn't set up yet. i did the changes online in github. maybe someone can run this for me while i try to set up my local github repo. thx :-)

Commit:
run generatecss.php

andrepereiradasilva commented:
I have tested this item :white_check_mark: successfully on 1c44b06d7d20a043d3e71030fd18027a56cc163a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9683">issues.joomla.org/joomla-cms/9683</a>.</sub>

brianteeman commented:
I have tested this item :white_check_mark: successfully on 1c44b06d7d20a043d3e71030fd18027a56cc163a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9683">issues.joomla.org/joomla-cms/9683</a>.</sub>

brianteeman commented:
RTC - thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9683">issues.joomla.org/joomla-cms/9683</a>.</sub>

JoomliC commented:
I have tested this item :red_circle: unsuccessfully on 1c44b06d7d20a043d3e71030fd18027a56cc163a<br /><br />Modal scrolling is broken on iphone and safari.
The reason i added overflow auto is due to a known issue with bootstrp 2 and 3. (Not fixed today) and because of a webkit issue.
The only known way to fix it is to force overflow.
I'm sorry, currently on holidays and with only my iphone, but i've alredy work on an improvement of my previous PR for 3.5.0
And could check it when back next week...
Sorry for timing...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9683">issues.joomla.org/joomla-cms/9683</a>.</sub>

brianteeman commented:
Removed RTC status<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9683">issues.joomla.org/joomla-cms/9683</a>.</sub>

infograf768 commented:
Imho, this should be added to blockers as batch is badly broken since 3.5.0

JoomliC commented:
@infograf768 i've worked on a redo of BS issue fix, and this one may solve batch issue too!
And the PR will be ready today (i'm testing again it, on different screen sizes and devices...)

Will inform about this PR in all issues related to the modals ;-)

JoomliC commented:
@infograf768 @ucfnet002 @brianteeman @wojsmol @andrepereiradasilva a PR is ready to be tested as a redo of PR 9140 to fix scrolling of Bootstrap modals on small devices and iOS as well as fix for batch dropdown issue: https://github.com/joomla/joomla-cms/pull/9817
;-)

brianteeman commented:
Closing in favour of #9817 Please test

zero-24 commented:
@brianteeman @rdeutz please remove the milestone as it get never merged
